front/scripts/i18n-checker.ts

```diff
@@ -166,98 +167,174 @@ async function getLocalesKeys(localePath: string, locale: string): Promise<Set<s
-          output: '$LOCALE/$NAMESPACE.json',
-        })
-      )
-      .pipe(vfs.dest(tempDir));
-
-    stream.on('finish', () => resolve(tempDir));
-    stream.on('error', (e: Error) => reject(e));
-  });
-}
+function visitCallExpression(
+  checker: ts.TypeChecker,
+  extractedKeys: Set<string>,
+  file: ts.SourceFile,
+  node: ts.CallExpression
+) {
+  const symbol = checker.getSymbolAtLocation(node.expression);
+  if (!symbol) {
+    return;
+  }
 
-/**
- * The script execution
- */
-async function run() {
-  try {
-    const keysByLocale = await Promise.all(
-      ['fr', 'en'].map(async (locale) => {
-        const keys = await getLocalesKeys(`${process.cwd()}/public/locales`, locale);
-        return { locale, keys };
-      })
-    );
+  const type = checker.getTypeOfSymbolAtLocation(symbol, node.expression);
+  if (type.symbol?.escapedName !== 'TFunction') {
+    return;
+  }
+  const typeArgs = checker.getTypeArguments(type as ts.TypeReference);
 
-    const scannedLocalePath = await scanCode();
-    const scannedNamespacedKeys = await getLocalesKeys(scannedLocalePath, 'dev');
+  if (typeArgs.length !== 2) {
+    throw new I18nSyntaxError(file, node, 'expected two generic type arguments for TFunction');
+  }
+  if (node.arguments.length < 1) {
+    throw new I18nSyntaxError(file, node, 'expected at least one argument for TFunction');
+  }
 
-    // Search for unused keys
-    const unusedKeys: Array<{ locale: string; key: string }> = [];
-    keysByLocale.forEach(({ locale, keys }) => {
-      keys.forEach((key) => {
-        if (
-          !scannedNamespacedKeys.has(key) &&
-          IGNORE_UNUSED.every((pattern) => !key.match(pattern))
-        ) {
-          unusedKeys.push({ locale, key });
-        }
-      });
-    });
+  const namespaceType = typeArgs[0];
+  const prefixType = typeArgs[1];
 
-    // Search for missing traduction
-    const missingKeys: Array<{ locale: string; key: string }> = [];
-    scannedNamespacedKeys.forEach((key) => {
-      if (IGNORE_MISSING.every((pattern) => !key.match(pattern))) {
-        keysByLocale.forEach(({ locale, keys }) => {
-          if (!keys.has(key)) {
-            missingKeys.push({ locale, key });
-          }
-        });
-      }
-    });
+  let defaultNamespace;
+  if (namespaceType.isStringLiteral()) {
+    defaultNamespace = namespaceType.value;
+  } else if (checker.isTupleType(namespaceType)) {
+    const namespaceTypeArgs = checker.getTypeArguments(namespaceType as ts.TypeReference);
+    if (!namespaceTypeArgs[0].isStringLiteral()) {
+      return;
+    }
+    defaultNamespace = namespaceTypeArgs[0].value;
+  } else {
+    return;
+  }
+
+  let prefix;
+  if (prefixType.isStringLiteral()) {
+    prefix = prefixType.value;
+  } else if (prefixType === checker.getUndefinedType()) {
+    prefix = null;
+  } else {
+    return;
+  }
+
+  const keyNode = node.arguments[0];
+  if (!ts.isStringLiteral(keyNode)) {
+    return;
+  }
+  let key = keyNode.text;
```

Comment:
These lines of code handle the key, while the code below (`optionsNode`) still handles the default namespace. 
Could you move these lines below the `optionsNode` handling code to make the logic easier to understand ?